### PR TITLE
[Go] Introduce gramework

### LINKED ISSUE #1135: Introduce gramework
+ Language : go
+ Repo : https://github.com/gramework/gramework
+ Author : @kirillDanshin

### PR DESCRIPTION
**Pull request by waghanza**
Hi,

This `PR` adds [gramework](https://github.com/gramework/gramework)

Closes #1135 

/cc @kirillDanshin

### CODE REVIEW

File: go/gramework/main.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"github.com/gramework/gramework"
+	"fmt"
+)
+
+func main() {
+	app := gramework.New()
+
+	app.GET("/", func(ctx *gramework.Context) error {
```
Comment (kirillDanshin) [inline go/gramework/main.go]: actually, in Gramework you can literally

```go
app.GET("/", "")
```

:)

File: go/gramework/main.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		_, err := ctx.WriteString("")
+		return err
+	})
+
+	app.GET("/user/:id", func(ctx *gramework.Context) error {
+                id := ctx.UserValue("id")
```
Comment (kirillDanshin) [inline go/gramework/main.go]: actually, I think you mean this:

```go
app.GET("/user/:id", func(ctx *gramework.Context) {
	ctx.SetBodyString(ctx.RouteArg("id"))
})
```

File: go/gramework/main.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		_, err := ctx.WriteString(fmt.Sprintf("%s", id))
+		return err
+        })
+
+	app.POST("/user", func(ctx *gramework.Context) error {
```
Comment (kirillDanshin) [inline go/gramework/main.go]: again,

```go
app.POST("/user", "")
```

:)

File: go/gramework/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+WORKDIR /go/src/app
+ADD . ./
+
+RUN go get
+RUN go build -a -o /go/bin/app ./
+
+FROM scratch
+COPY --from=buildenv /go/bin/app /go/bin/app
+CMD ["/go/bin/app"]
```
Comment (kirillDanshin) [inline go/gramework/Dockerfile]: Please, add ENV=production. By default, gramework logs too much for production usage, but enough for development convenience

**Comment by kirillDanshin** [comment]
thanks!

File: go/gramework/main.go (hunk 1 above)
Comment (waghanza) [reply go/gramework/main.go]: :heart:

File: go/gramework/main.go (hunk 2 above)
Comment (waghanza) [reply go/gramework/main.go]: :heart:

File: go/gramework/main.go (hunk 3 above)
Comment (waghanza) [reply go/gramework/main.go]: :heart:

File: go/gramework/Dockerfile (hunk 4 above)
Comment (waghanza) [reply go/gramework/Dockerfile]: ok

**Comment by waghanza** [comment]
@kirillDanshin I fix all issues. Thanks for advices, is the source ok for you now ?

**Comment by kirillDanshin** [comment]
pretty much ok, except I'd rather use `app.ListenAndServe()` without arguments in real projects, and setup the port using `PORT=3000` or built-in `-bind :3000`, which defaults to `:80`

thanks for review request, I'm looking forward for benchmark results!

**Comment by waghanza** [comment]
@kirillDanshin all's good now?

**Comment by kirillDanshin** [comment]
@waghanza yes, thanks :)

**Comment by waghanza** [comment]
@kirillDanshin

**Comment by kirillDanshin** [comment]
@waghanza?

**Comment by waghanza** [comment]
@kirillDanshin sorry, mis-clicked

I've updated results (I have reduced `docker` **noise**) but still need improvements ^^

**Comment by kirillDanshin** [comment]
@waghanza well, I see very low RPS and high latency. looks strange. do you configure the kernel before testing?

**Comment by waghanza** [comment]
@kirillDanshin this is a side effect of `local` **docker**
the results will be updated regularly

**Pull request merged**